caddy: stop using xcaddy --with flag to build

hairyhenderson opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Fixes #61621 - addresses the panic by moving away from the bad invocation of `xcaddy`, and using it correctly instead.

Now the resulting version number is correct:

```console
$ caddy version
v2.2.0 h1:sMUFqTbVIRlmA8NkFnNt9l7s0e+0gw+7GPIrhty905A=
```

Signed-off-by: Dave Henderson <dhenderson@gmail.com>

mholt reviewed (changes requested):
When you get the chance, it'd be nice to remove the `--resume` flag as well; I assume most users will be using a Caddyfile. We should eliminate _either_ the `--resume` flag or the `--config` flag.

hairyhenderson commented:
Ok I adjusted my original approach a bit, and simply went back to using `xcaddy` to build. I also removed `--resume` as it's spurious, and pinned to Go 1.15 (so we can upgrade to 1.16 on our own schedule in case there's another bug with a dependency like the one that forced us to pin to 1.14).

SMillerDev commented on Formula/caddy.rb:
```diff
@@ -13,7 +13,7 @@ class Caddy < Formula
   end
 
-  depends_on "go" => :build
+  depends_on "go@1.15" => :build
```
This should just be `go`. When go 1.16 comes out we'll test the formula and if it doesn't work we'll pin it to 1.15.

francislavoie replied:
Why not just let us tell you when it's good to use?

The version was pinned to use Go 1.14 earlier because there was a breakage in a dependency of Caddy in relation to Go 1.15, so that dependency had to be upgraded before it Go 1.15 would work.

There's no guarantee that the same won't happen for 1.16. Pinning is the safer bet. Let us first validate in the upstream repo when we're ready to have builds use the latest version of Go, then brew can be updated to the latest.

SMillerDev replied:
Because homebrew is an evergreen package manager. We always use the latest dependencies that work and since your software is open source the Homebrew project can decide to package it whatever way it likes.

francislavoie replied:
The problem is that we're responsible as maintainers of the project when things break. Users will come to us to complain.

This entire situation happened because changes were made here without getting our input, and we heard about the breakage via our forums.

I'd be much happier if things were played safe here, and the software was built in a predictable way, consistent with how it's built everywhere else.

SMillerDev replied:
You can make the software closed source and have full control over the way you distribute it. Until then you'll have to deal with the best practices that package managers are using for themselves.

Some information about Homebrew's position on upstream control of packages: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap#official-vendor-taps

hairyhenderson replied:
I'm fine with this being just `go`. Updating...

SMillerDev reviewed (approved):
Thanks @hairyhenderson  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉
